src/neighbors.cc

```diff
@@ -571,7 +524,7 @@ neighbors_init(void)
                 debugs(15, DBG_IMPORTANT, "WARNING: Peer looks like this host." <<
                        Debug::Extra << "Ignoring cache_peer " << *thisPeer);
 
-                neighborRemove(thisPeer);
+                NeighborRemove(thisPeer.get());
```

Comment:
Does the official code crash or trigger UB if the inner/HttpPortList loop continues after a neighborRemove() call? The current PR code has that problem, but I want to understand whether we are introducing a new bug here or preserving an old one.

This change request is _unrelated_ to the "removing items from the being-iterated vector" question in another change request.